src/Framework/StringUtils.cs

```diff
@@ -17,15 +18,16 @@ internal static class StringUtils
     internal static string GenerateRandomString(int length)
     {
         // Base64, 2^6 = 64
+        using var rng = RandomNumberGenerator.Create();
```

Comment:
Is there a performance advantage in holding one of these in a static (or thread static)? I don't know, but some implementations of such things generate a block and then return it in pieces as requested. Or pull expensive entropy to start up. @vcsjones  ?

Reply:
`RandomNumberGenerator` was never documented as thread-safe, even though in all practical cases, it is thread-safe when used with `RandomNumberGenerator.Create()`. Practically speaking, it would work fine to pull this up to a static, even in .NET Framework.

We've made significant investments in the RandomNumberGenerator for .NET (Core). If you want to avoid the allocation for the modern .NET, you can do something like

```C#
byte[] randomBytes;

#if NET
  randomBytes = RandomNumberGenerator.GetBytes(randomBytesNeeded);
#else
  randomBytes = new byte[randomBytesNeeded];
  using var rng = RandomNumberGenerator.Create();
  rng.GetBytes(randomBytes);
#endif
```

Reply:
Thank you for suggestions.

We're currently using this utility to create name of a build binary log file - so simpler code is preference